File: backend/models/project.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from bson import ObjectId

class ProjectBase(BaseModel):
    name: str
    description: Optional[str] = None
    icon: Optional[str] = "📁"
    color: Optional[str] = "#0086c0"
    owner: str
    members: List[str] = []
    department: Optional[str] = None
    startDate: Optional[datetime] = None
    endDate: Optional[datetime] = None
    budget: Optional[float] = None
    status: str = "planning"  # planning, in_progress, on_hold, completed, cancelled
    priority: str = "medium"  # low, medium, high, urgent
    favorite: bool = False

class ProjectCreate(ProjectBase):
    pass

class ProjectUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    icon: Optional[str] = None
    color: Optional[str] = None
    members: Optional[List[str]] = None
    department: Optional[str] = None
    startDate: Optional[datetime] = None
    endDate: Optional[datetime] = None
    budget: Optional[float] = None
    status: Optional[str] = None
    priority: Optional[str] = None
    favorite: Optional[bool] = None

class Project(ProjectBase):
    id: str = Field(default_factory=lambda: str(ObjectId()), alias="_id")
    createdBy: str
    createdAt: datetime = Field(default_factory=datetime.utcnow)
    updatedAt: datetime = Field(default_factory=datetime.utcnow)

    class Config:
        populate_by_name = True
        arbitrary_types_allowed = True
        json_encoders = {ObjectId: str}
        json_schema_extra = {
            "example": {
                "name": "Web Sitesi Yenileme",
                "description": "Şirket web sitesinin yenilenmesi",
                "icon": "🌐",
                "color": "#0086c0",
                "status": "in_progress",
                "priority": "high"
            }
        }

File: dotnet-backend/Unity.Core/Models/Project.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json;

namespace Unity.Core.Models
{
    public class Project
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        public string? Description { get; set; }
        public string Icon { get; set; } = "📁";
        public string Color { get; set; } = "#0086c0";
        public int Owner { get; set; } // UserId
        
        // Stored as JSON
        public string MembersJson { get; set; } = "[]";

        [NotMapped]
        public List<int> Members 
        { 
            get => JsonSerializer.Deserialize<List<int>>(MembersJson ?? "[]") ?? new List<int>();
            set => MembersJson = JsonSerializer.Serialize(value);
        }

        public int DepartmentId { get; set; } // Changed from string Department name
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public double? Budget { get; set; }
        
        public string Status { get; set; } = "planning"; // planning, in_progress, on_hold, completed, cancelled
        public string Priority { get; set; } = "medium"; // low, medium, high, urgent
        
        public bool Favorite { get; set; } = false;
        public bool IsPrivate { get; set; } = false;

        public int CreatedBy { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
